### DISCUSSION TITLE
Conditionally add logger middleware

### DISCUSSION THREAD

**Discussion by chemmanouilidis** [reduxjs/redux-toolkit#1546] [Q&A]
Hello,

What is the most efficient/readable way of adding logger based on process_env variable? I tried to come up with some workarounds but they were either really nasty to read or using the deprecated getDefaultMiddleware() function.

**Comment by phryneas** [comment]
`getDefaultMiddleware` is only deprecated as an import, you can still write code with the callback notation of 
```js
middleware: getDefaultMiddleware => {
// your code here
}
```

Personally I'd just recommend to use the DevTools instead of the logger though - for me, that spam makes it impossible to see actually important info on the console.

**Reply by markerikson** [reply]
Yeah, I basically see `redux-logger` as being useless given the existence of the DevTools. I suppose there might be rare use cases for it, but I sure haven't ever needed it in any of the work I've done.
